src/UglyToad.PdfPig/Graphics/ContentStreamProcessor.cs

```diff
@@ -603,6 +603,12 @@ private void ProcessFormXObject(StreamToken formStream)
             // 3. We don't respect clipping currently.
 
             // 4. Paint the objects.
+            if (parsingOptions.UseLenientParsing && xObjectName != null && operations.OfType<InvokeNamedXObject>()?.Any(o => o.Name == xObjectName) == true)
```

Comment:
Would this still stack overflow when `parsingOptions.UseLenientParsing` is `false`? Maybe the other branch of this `if` should throw an exception in that case?

Reply:
it would yes,and makes sense to throw an exception. Will update that

Reply:
I need to do further investigation in fact. MOZILLA-3136-0.pdf is flagged as having circular reference now, but was never an issue with that (it seems)

Reply:
I've updated the logic to also check the forms data